### DISCUSSION TITLE
Le Potato and UGS

### DISCUSSION THREAD

**Discussion by evanfeld33** [winder/Universal-G-Code-Sender#2169] [Help]
I am trying to run UGS on AML-S905x-CC (Le Potato) because of the raspberry pi shortage.

I have flashed the raspbian image (2022-09-22-raspbian-bullseye-arm64+aml-s905x-cc.img.xz) from the potato’s website.

I am running the potato on a touchscreen from Amazon and I have downloaded the Latest release (v2.0.14) of UGS for raspberry pi. 

My issue is when I try to open or run the ugsplatform file, the application fails to open. There is no error code.

Any help is appreciated.

**Comment by breiler** [comment]
Can you please attach the log file from: ~/.ugsplatform/var/log/messages.log

**Comment by evanfeld33** [comment]
> Can you please attach the log file from: ~/.ugsplatform/var/log/messages.log

Hello,
I apologize for my lack of knowledge, but I cannot find the file you reference. Attached are the folders that I see. ![image](https://user-images.githubusercontent.com/127077919/223312020-dc9a5c02-1fbb-47e9-8a02-15c29342d4aa.jpeg)

**Comment by evanfeld33** [comment]
> Can you please attach the log file from: ~/.ugsplatform/var/log/messages.log

Hello,
When I try to open up the ugsplatform file, a terminal window pops up for a split second then closes. I recorded the process and screenshot one of the frames. It gives an error message - to me it seems like the Java package might not have downloaded correctly.

I attempted to redownload Java by executing 
"Sudo apt-get install openjdk-8-jdk"
The terminal said "unable to locate package". I retried with 
Openjdk-11-jdk, which allowed me to install.

But even after this, I cannot get ugs to open. ![image](https://user-images.githubusercontent.com/127077919/223320221-892465dc-c41d-4631-8120-fee157743df1.png)

**Comment by breiler** [comment]
Ahh, you need a 32-bit OS on ARM to be able to run UGS. 
I could never get it to run properly on a ARMx64.

See the section for rasberry pi here:
https://github.com/winder/Universal-G-Code-Sender/wiki/Installing

**Reply by evanfeld33** [reply]
Hello,
Okay, so i looked into the Le Potato images and found one that is supposed to be 32-bit. Found here: https://hub.libre.computer/t/raspbian-10-buster-for-libre-computer-boards/415

Upon opening UGS i get the attached error log. Any idea what could be the issue? thanks for your help on this.
[hs_err_pid6579.log](https://github.com/winder/Universal-G-Code-Sender/files/10917457/hs_err_pid6579.log)

**Reply by evanfeld33** [reply]
and to clarify, i still cannot get UGS to open

**Reply by breiler** [reply]
The logs did not give any clues, sorry I can't help you.

**Comment by evanfeld33** [comment]
Just wanted to document the solution, in case anybody else has this issue.

I flashed the Raspian 10 version to LePotato, since it is supposedly 32 bit.

After attempting again to download UGS for raspberry pi - I kept getting different errors and had errors logs that I couldn’t decipher) So what I did was instead of downloading the raspberry pi version of UGS, I downloaded the “all platforms” version (2.0.14).

I almost gave up, but after a bit of searching on the web, all my issues seemed to be related to a Java mismatch.

I reinstalled Java11 but still got errors.After that, I installed Java 8. 

Since I had multiple versions of Java installed, I had to make sure that Java 8 was the selected version.

And after that… success! I have UGS running and connected to my MKS GEN L.
